core/security_group_rule.go

```diff
@@ -0,0 +1,188 @@
+package core
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/krystal/go-katapult"
+)
+
+type SecurityGroupRule struct {
+	ID        string    `json:"id,omitempty"`
+	Direction string    `json:"direction,omitempty"`
+	Protocol  string    `json:"protocol,omitempty"`
+	Ports     *string   `json:"ports,omitempty"`
+	Targets   *[]string `json:"targets,omitempty"`
+	Notes     *string   `json:"notes,omitempty"`
```

Comment:
For this struct these fields actually don't need to be pointer types, as it's used to hold rule information received from the API. It's only the `SecurityGroupRuleArguments` struct where they need to be pointers so we can differentiate in JSON between not sending the key at all or explicitly setting it to empty string/array when creating/updating rules.